kinesis/src/main/scala/akka/stream/alpakka/kinesis/KinesisSchedulerSettings.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.kinesis
+
+import scala.concurrent.duration._
+
+final class KinesisSchedulerSourceSettings private (val bufferSize: Int, val backpressureTimeout: FiniteDuration) {
+  require(
+    bufferSize >= 1,
+    "Buffer size must be greater than 0; use size 1 to disable stage buffering"
+  )
+  def withBufferSize(size: Int): KinesisSchedulerSourceSettings =
+    KinesisSchedulerSourceSettings(size, backpressureTimeout)
+  def withBackpressureTimeout(timeout: java.time.Duration): KinesisSchedulerSourceSettings =
+    KinesisSchedulerSourceSettings(bufferSize, timeout)
+}
+final class KinesisSchedulerCheckpointSettings private (val maxBatchSize: Int, val maxBatchWait: FiniteDuration) {
+  require(
+    maxBatchSize >= 1,
+    "Batch size must be greater than 0; use size 1 to commit records one at a time"
+  )
+
+  def withMaxBatchSize(size: Int): KinesisSchedulerCheckpointSettings =
+    KinesisSchedulerCheckpointSettings(size, maxBatchWait)
+  def withMaxBatchWait(duration: java.time.Duration): KinesisSchedulerCheckpointSettings =
+    KinesisSchedulerCheckpointSettings(maxBatchSize, duration)
+}
+
+object KinesisSchedulerSourceSettings {
+
+  def apply(bufferSize: Int, backpressureTimeout: FiniteDuration): KinesisSchedulerSourceSettings =
+    new KinesisSchedulerSourceSettings(bufferSize, backpressureTimeout)
+  def apply(bufferSize: Int, backpressureTimeout: java.time.Duration): KinesisSchedulerSourceSettings =
+    KinesisSchedulerSourceSettings(bufferSize, FiniteDuration.apply(backpressureTimeout.toMillis, MILLISECONDS))
+  def apply: KinesisSchedulerSourceSettings = KinesisSchedulerSourceSettings(1000, 1.minute)
+
+  val defaultInstance: KinesisSchedulerSourceSettings = KinesisSchedulerSourceSettings.apply
+
+  /**
+   * Java API
+   */
+  def create(bufferSize: Int, backpressureTimeout: FiniteDuration): KinesisSchedulerSourceSettings =
+    new KinesisSchedulerSourceSettings(bufferSize, backpressureTimeout)
```

Comment:
No reason for `java.time.Duration` in `apply`, but in the Java API.
You should create the instance in `val defaultInstance` and return that from `apply`.